### PR TITLE
HZN-879: location specific snmp-config

### PR DESCRIPTION
**Pull request by cgorantla**
Adding an attribute location ( Minion location ) to Definition 
Modify DAO to load it.

* JIRA: http://issues.opennms.org/browse/hzn-879
* Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS980

### CODE REVIEW

File: opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/SnmpConfigVisitor.java [hunk 1]
```diff
@@ -45,4 +45,6 @@
     
     void visitIpMatches(List<String> ipMatches);
     void visitIpMatchesFinished();
+
+    String getLocation();
```
Comment (j-white) [inline opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/SnmpConfigVisitor.java]: This should not be part of the visitor interface.

File: opennms-config-jaxb/src/main/resources/xsds/snmp-config.xsd [hunk 2]
```diff
@@ -248,7 +248,14 @@
                 <documentation>Match Octets (as in IPLIKE)</documentation>
               </annotation>
             </element>
-          </sequence>
+		  </sequence>
```
Comment (j-white) [inline opennms-config-jaxb/src/main/resources/xsds/snmp-config.xsd]: Can we fix the indentation to match the existing document?

File: opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/Definition.java [hunk 3]
```diff
@@ -184,10 +197,23 @@ public boolean equals(final Object obj) {
...
+        } else if (!m_location.equals(other.m_location)) {
+            return false;
+        }
+
         return true;
     }
 
     public void visit(final SnmpConfigVisitor visitor) {
+
+        if ((visitor.getLocation() != null) && (!visitor.getLocation().equals(m_location))) {
```
Comment (j-white) [inline opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/Definition.java]: This logic is in the wrong place, it should be up to the visitor implementation to decide if it wants to visit a definition or not.

File: opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/SnmpConfigVisitor.java (hunk 1 above)
Comment (cgorantla) [inline opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/SnmpConfigVisitor.java]: done

File: opennms-config-jaxb/src/main/resources/xsds/snmp-config.xsd (hunk 2 above)
Comment (cgorantla) [inline opennms-config-jaxb/src/main/resources/xsds/snmp-config.xsd]: done

File: opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/Definition.java (hunk 3 above)
Comment (cgorantla) [inline opennms-config-jaxb/src/main/java/org/opennms/netmgt/config/snmp/Definition.java]: removed this logic

**Comment by j-white** [comment]
Looks good, waiting on Bamboo.

**Pull request merged**